runtime/compiler/optimizer/J9RecognizedCallTransformer.hpp

```diff
@@ -139,6 +139,88 @@ class RecognizedCallTransformer : public OMR::RecognizedCallTransformer
     *     \endcode
     */
    void process_java_lang_StringUTF16_toBytes(TR::TreeTop* treetop, TR::Node* node);
+   /** \brief
+    *     Transforms jdk/internal/util/ArraysSupport.vectorizedMismatch(Ljava/lang/Object;JLjava/lang/Object;JII)I
+    *     into an arraycmp, bit manipulation and iselect sequence with equivalent semantics.
+    *
+    *  \param treetop
+    *     The treetop which anchors the call node.
+    *
+    *  \param node
+    *     The call node representing a call to jdk/internal/util/ArraysSupport.vectorizedMismatch(Ljava/lang/Object;JLjava/lang/Object;JII)I
+    *     which has the following shape:
+    *
+    *     \code
+    *     icall  <jdk/internal/util/ArraysSupport.vectorizedMismatch(Ljava/lang/Object;JLjava/lang/Object;JII)I>
+    *       <a>
+    *       <aOffset>
+    *       <b>
+    *       <bOffset>
+    *       <length>
+    *       <log2ArrayIndexScale>
+    *     \endcode
+    *
+    *  \details
+    *     The call node is transformed to the following shape:
+    *
+    *     \code
+    *     iselect ()
+    *       lcmpeq
+    *         arraycmplen
+    *           aladd
+    *             <a>
+    *             <aOffset>
+    *           aladd
+    *             <b>
+    *             <bOffset>
+    *           land
+    *             lshl
+    *               i2l
+    *                 <length>
+    *               i2l
+    *                 <log2ArrayIndexScale>
+    *             lxor
+    *               lor
+    *                 lshl
+    *                   ==>i2l
+    *                   lconst 1
+    *                 lconst 3
+    *               lconst -1
+    *         ==>land
+    *       ixor
+    *         l2i
+    *           lshr
+    *             land
+    *               ==>lshl
+    *               ==>lor
+    *             ==>i2l
+    *         iconst -1
+    *       ishr
```

Comment:
This should be `l2i` of `lshr` rather than `ishr`. The actual transformation is correct